src/ring_theory/unique_factorization_domain.lean

```diff
@@ -545,6 +549,16 @@ begin
   { exact dvd_trans (multiset.dvd_prod H) (associated.dvd (normalized_factors_prod hcases)) },
 end
 
+lemma associated_prime_pow_of_unique_normalized_factor {p r : α}
+  (h : ∀ {m}, m ∈ normalized_factors r → m = p) (hr : r ≠ 0) : ∃ (i : ℕ), associated (p^i) r :=
+begin
+  use (normalized_factors r).card,
+  have hi : normalized_factors r = multiset.repeat p (normalized_factors r).card,
+  { exact multiset.eq_repeat_of_mem (λ b hb, h hb) },
+  have := unique_factorization_monoid.normalized_factors_prod hr,
+  rwa [hi, multiset.prod_repeat] at this
```

Comment:
```suggestion
  use (normalized_factors r).card,
  have := unique_factorization_monoid.normalized_factors_prod hr,
  rwa [multiset.eq_repeat_of_mem (λ b, h), multiset.prod_repeat] at this
```

I don't mind the existential but you could also just write `associated (p ^ (...).card) r`. whatever you feel is more useful!